ted_sws/metadata_normaliser/services/xml_manifestation_metadata_extractor.py

```diff
@@ -0,0 +1,279 @@
+from typing import List
+import xml.etree.ElementTree as ET
+from io import StringIO
+
+from ted_sws.domain.model.manifestation import XMLManifestation
+from ted_sws.metadata_normaliser.model.metadata import ExtractedMetadata, LanguageTaggedString, CompositeTitle, \
+    EncodedValue
+from ted_sws.metadata_normaliser.services.xpath_registry import XpathRegistry
+
+
+def extract_text_from_element(element: ET.Element) -> str:
+    """
+    Extract text from an element in the XML structure
+    :param element:
+    :return: str
+    """
+    if element is not None:
+        return element.text
+
+
+def extract_attribute_from_element(element: ET.Element, attrib_key: str) -> str:
+    """
+    Extract attribute value from an element in the XML structure
+    :param element:
+    :param attrib_key:
+    :return:
+    """
+    if element is not None:
+        return element.attrib[attrib_key]
+
+
+def extract_code_and_value_from_element(element: ET.Element) -> EncodedValue:
+    """
+    Extract code attribute and text values from an element in the XML structure
+    :param element:
+    :return:
+    """
+    if element is not None:
+        return EncodedValue(code=extract_attribute_from_element(element=element, attrib_key="CODE"),
```

Comment:
Warning: we might not always have "CODE" attribute; it is worth having it as a default variable.